### PR TITLE
[Terraform]  Editing data-sources/ec_stack.md

### PR DESCRIPTION
**Pull request by alaudazzi**
## Description
This PR proposes some editing suggestions.

## Related Issues
#78

## Notes for the reviewers

- Capitalize the word "stack" only in combination with Elastic
- EC (SaaS) => Elastic Cloud ESS?
- In the Attribute Reference, double check my suggestions for: 
   * `accessible` - To have this version accessible/not accessible by the calling user. This is only relevant for Elastic Cloud ESS, not for ECE.
   * `min_upgradable_from` - The minimum stack version recommended.
   * `upgradable_to` - The stack version you can upgrade to.
   * `allowlisted` - To include/not include this version in the `allowlist`. This is only relevant for Elastic Cloud ESS, not for ECE.

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improves code quality but has no user-facing effect)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation

### CODE REVIEW

**Review by marclop — APPROVED** [review]
LGTM

File: docs/data-sources/ec_stack.md [hunk 1]
```diff
@@ -24,41 +24,41 @@ data "ec_stack" "latest_patch" {
...
 
 ~> **NOTE:** Depending on the stack version, some values may not be set. These will not be available for interpolation.
 
 * `version` - The stack version.
-* `accessible` - Whether or not this version is accessible by the calling user. This is only relevant in EC (SaaS) and is not sent in ECE.
-* `min_upgradable_from` - The minimum version recommended to upgrade to this version.
-* `upgradable_to` - The Stack versions that this version can upgrade to.
-* `allowlisted` - Whether or not this version is in the allowlist. This is only relevant in EC (SaaS) and is not sent in ECE.
-* `apm` - Information for APM workloads on this Stack version.
+* `accessible` - To have this version accessible/not accessible by the calling user. This is only relevant for Elastic Cloud ESS, not for ECE.
```
Comment (karencfv) [inline docs/data-sources/ec_stack.md:37]: In other parts of the documentation we refer to it as "Elasticsearch Service", should we keep it here as well? Or should we change the other ones?

```suggestion
* `accessible` - To have this version accessible/not accessible by the calling user. This is only relevant for Elasticsearch Service (ESS), not for ECE.
```

File: docs/data-sources/ec_stack.md [hunk 2]
```diff
@@ -24,41 +24,41 @@ data "ec_stack" "latest_patch" {
...
+* `min_upgradable_from` - The minimum stack version recommended.
+* `upgradable_to` - The stack version you can upgrade to.
+* `allowlisted` - To include/not include this version in the `allowlist`. This is only relevant for Elastic Cloud ESS, not for ECE.
```
Comment (karencfv) [inline docs/data-sources/ec_stack.md:40]:
```suggestion
* `allowlisted` - To include/not include this version in the `allowlist`. This is only relevant for Elasticsearch Service (ESS), not for ECE.
```

**Comment by alaudazzi** [comment]
@elasticmachine update branch

**Pull request merged**